### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by CodaFi**

### CODE REVIEW

**Review by segiddins — APPROVED** [review]
Looks amazing!

File: README.md [hunk 1]
```diff
@@ -1,53 +1,167 @@
...
+Defining a simple function and moving the cursor to a point where we can begin inserting instructions is done like so:
 
 ```swift
-let main = builder.addFunction(name: "main", 
-                               type: FunctionType(argTypes: [],
-                                                  returnType: VoidType()))
+let builder = IRBuilder(module: module)
+
+let main = builder.addFunction(
+							name: "main",
```
Comment (harlanhaskins) [inline README.md]: Tabs -> spaces

File: README.md [hunk 2]
```diff
@@ -1,53 +1,167 @@
...
 ```swift
-let main = builder.addFunction(name: "main", 
-                               type: FunctionType(argTypes: [],
-                                                  returnType: VoidType()))
+let builder = IRBuilder(module: module)
+
+let main = builder.addFunction(
+							name: "main", 
+                            type: FunctionType(argTypes: [],
+                           	returnType: VoidType())
```
Comment (harlanhaskins) [inline README.md]: Stray tab

File: README.md [hunk 3]
```diff
@@ -1,53 +1,167 @@
...
+  }
+  return retVal
+}
+```
 
-Plus, it provides common wrappers around oft-used types like `Function`, `Global`, `Switch`, and `PhiNode`.
+Notice that the value of `retVal` depends on the path the flow of control takes through this program, so we must emit a Phi node to properly initialize it:
+
+```swift
+let function = builder.addFunction("calculateFibs", type: FunctionType(argTypes: [IntType.int1], returnType: FloatType.double))
```
Comment (harlanhaskins) [inline README.md]: Let's align these parameters

Comment (harlanhaskins) [inline README.md]: Should we explain that booleans are `int1`?

File: README.md [hunk 4]
```diff
@@ -1,53 +1,167 @@
...
+}
+```
+
+### JIT
+
+LLVMSwift provides a JIT abstraction to make executing code in LLVM modules quick and easy.  Let's execute the PHI node example from before:
+
+```swift
+// Setup the JIT
+let jit = try! JIT(module: module, machine: TargetMachine())
```
Comment (harlanhaskins) [inline README.md:174]: Pretty sure the machine is defaulted, no?

**Review by harlanhaskins — CHANGES_REQUESTED** [review]

**Comment by harlanhaskins** [comment]
Looks amazing modulo nitpicks!

File: README.md (hunk 4 above)
Comment (CodaFi) [reply README.md:174]: It is not.

File: README.md [hunk 5]
```diff
@@ -1,46 +1,160 @@
...
+  }
+  return retVal
+}
+```
 
-Plus, it provides common wrappers around oft-used types like `Function`, `Global`, `Switch`, and `PhiNode`.
+Notice that the value of `retVal` depends on the path the flow of control takes through this program, so we must emit a PHI node to properly initialize it:
+
+```swift
+let function = builder.addFunction("calculateFibs", type: FunctionType(argTypes: [IntType.int1], returnType: FloatType.double))
```
Comment (harlanhaskins) [inline README.md]: Still maybe align these parameters?

**Review by harlanhaskins — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (CodaFi) [reply README.md]: I did want to add a section on types originally.  I'll see what I can do.

**Comment by CodaFi** [comment]
New section on types

**Review by harlanhaskins — APPROVED** [review]

**Comment by harlanhaskins** [comment]
🚢 :it:

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**